adaptive/runner.py

```diff
@@ -198,20 +212,22 @@ def overhead(self):
     def _process_futures(self, done_futs):
         for fut in done_futs:
             x = self.pending_points.pop(fut)
+            i = _key_by_value(self._index_to_point, x)  # O(N)
```

Comment:
In general, we can expect that the very first value in `self._index_to_point` will be `x` because it is simply the oldest/longest running point.

So I really don't think this operation will hurt.

@jbweston, do you have an opinion here?

Reply:
If we change `pending_points` to store point IDs instead then we don't need this reverse lookup at all.